MatrixSDK/VoIP/MXCallManager.m

```diff
@@ -1267,6 +1387,199 @@ - (void)placeCallAgainst:(NSString *)phoneNumber
+            {
+                //  Wait for other party to join before returning the room
+                __block BOOL joined = NO;
+                
+                __block id listener = [room listenToEventsOfTypes:@[kMXEventTypeStringRoomMember] onEvent:^(MXEvent *event, MXTimelineDirection direction, MXRoomState *roomState) {
+                    if ([event.sender isEqualToString:userId])
+                    {
+                        MXRoomMemberEventContent *content = [MXRoomMemberEventContent modelFromJSON:event.content];
+                        if (content.membership == kMXMembershipStringJoin)
+                        {
+                            joined = YES;
+                            [room removeListener:listener];
+                            dispatch_async(dispatch_get_main_queue(), ^{
+                                completion(room, nil);
+                            });
+                        }
+                    }
+                }];
+                
+                //  implement the timeout
+                dispatch_after(dispatch_time(DISPATCH_TIME_NOW, timeout * NSEC_PER_SEC), dispatch_get_main_queue(), ^{
+                    if (!joined)
+                    {
+                        //  user failed to join within the given time
+                        completion(nil, nil);
+                    }
+                });
+            }
+            else
+            {
+                completion(nil, nil);
+            }
+        }];
+    }
+    else
+    {
+        //  we're not in a direct room with target, create it
+        [self.mxSession canEnableE2EByDefaultInNewRoomWithUsers:@[userId] success:^(BOOL canEnableE2E) {
+            
+            MXRoomCreationParameters *roomCreationParameters = [MXRoomCreationParameters parametersForDirectRoomWithUser:userId];
+            roomCreationParameters.visibility = kMXRoomDirectoryVisibilityPrivate;
+            roomCreationParameters.creationContent = [MXRoomCreationParameters creationContentForVirtualRoomWithNativeRoomId:nativeRoomId];
+            
+            if (canEnableE2E)
+            {
+                roomCreationParameters.initialStateEvents = @[
+                    [MXRoomCreationParameters initialStateEventForEncryptionWithAlgorithm:kMXCryptoMegolmAlgorithm]
+                ];
+            }
+
+            [self.mxSession createRoomWithParameters:roomCreationParameters success:^(MXRoom *room) {
+                //  set account data on the room
+                MXWeakify(room);
+                [room setAccountData:@{
+                    kRoomNativeRoomIdJSONKey: nativeRoomId
+                } forType:kRoomIsVirtualJSONKey success:^{
+                    //  wait for other party to join
+                    MXStrongifyAndReturnIfNil(room);
+                    [self.mxSession setVirtualRoom:room.roomId forNativeRoom:nativeRoomId];
+                    return [self directCallableRoomWithUser:userId timeout:timeout completion:completion];
```

Comment:
I am surprised that this `return` compiles.
Is it me or is there an issue?

Reply:
It was redundant, removed.